packages/mui-material/src/StepContent/StepContent.js

```diff
@@ -83,24 +86,33 @@ const StepContent = React.forwardRef(function StepContent(inProps, ref) {
     transitionDuration = undefined;
   }
 
+  const externalForwardedProps = {
+    slots: { transition: TransitionComponent, ...slots },
```

Comment:
This should be:

```suggestion
    slots,
```

As `TransitionComponent` is being passed through `as`